disnake/guild_scheduled_event.py

```diff
@@ -0,0 +1,372 @@
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+
+from __future__ import annotations
+
+from datetime import datetime
+from typing import Any, Dict, List, Optional, TYPE_CHECKING, Union
+from .enums import (
+    GuildScheduledEventEntityType,
+    GuildScheduledEventStatus,
+    StagePrivacyLevel,
+    try_enum,
+)
+from .user import User
+from .member import Member
+from .mixins import Hashable
+from .utils import cached_slot_property, parse_time, _get_as_snowflake, MISSING
+
+if TYPE_CHECKING:
+    from .abc import GuildChannel
+    from .guild import Guild
+    from .state import ConnectionState
+
+
+__all__ = ("GuildScheduledEventMetadata", "GuildScheduledEvent")
+
+
+class GuildScheduledEventMetadata:
+    """
+    Represents guild event entity metadata.
+
+    .. versionadded:: 2.3
+
+    Attributes
+    ----------
+    location: Optional[:class:`str`]
+        Location of the event
+    """
+
+    __slots__ = ("location",)
+
+    def __init__(self, *, location: str = None):
+        self.location: Optional[str] = location
+
+    def to_dict(self) -> Dict[str, Any]:
+        return {"location": self.location}
+
+    @classmethod
+    def from_dict(cls, data):
+        return GuildScheduledEventMetadata(location=data.get("location"))
+
+
+class GuildScheduledEvent(Hashable):
+    """
+    Represents guild scheduled events.
+
+    .. versionadded:: 2.3
+
+    .. container:: operations
+
+        .. describe:: x == y
+
+            Checks if two scheduled events are equal.
+
+        .. describe:: x != y
+
+            Checks if two scheduled events are not equal.
+
+        .. describe:: hash(x)
+
+            Returns the scheduled event's hash.
+
+    Attributes
+    ----------
+    id: :class:`int`
+        The id of the scheduled event.
+    guild_id: :class:`int`
+        The guild id which the scheduled event belongs to.
+    channel_id: Optional[:class:`int`]
+        The channel id in which the scheduled event will be hosted.
+        This field is ``None`` if :attr:`entity_type` is :class:`GuildScheduledEventEntityType.external`
+    creator_id: Optional[:class:`int`]
+        The id of the user that created the scheduled event.
+        This field is ``None`` for events created before October 25th, 2021.
+    name: :class:`str`
+        The name of the scheduled event (1-100 characters).
+    description: :class:`str`
+        The description of the scheduled event (1-1000 characters).
+    scheduled_start_time: :class:`datetime`
+        The time the event will start.
+    scheduled_end_time: Optional[:class:`datetime`]
+        The time the event will end, or ``None`` if the event does not have a scheduled time to end.
+    privacy_level: :class:`StagePrivacyLevel`
```

Comment:
This should be its own type, scheduled events currently only have one privacy level (only `2`, `1` was removed).

Changing this everywhere also requires introducing a new audit log transformer for `privacy_level`.